### ISSUE TITLE
WP Job Manager compatibility

### ISSUE THREAD

**Issue by ethanclevenger91** [elementor/elementor#8154]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
Currently, the WP Job Manager posts archive doesn't work with Elementor. Similarly, WP Job Manager posts' custom fields can't be fetched via Elementor's custom field connection feature.

**Describe the solution you'd like**
For the post type archive compatibility, your query to determine valid post types includes the following arg:

`array ( 'show_in_nav_menus' => true, )`

Seems like a roundabout way of determining public post types. Seems the following would be more appropriate:

`array ('public' => true, )`

But I don't know the intricacies of that. Alternatively, a change would have to happen on WP Job Manager's end with how they register the post type. There's [an open issue](https://github.com/Automattic/WP-Job-Manager/issues/1613) there describing all this as well.

For the custom fields, WP Job Manager prefixes its meta keys with underscores. For obvious reasons, ignoring such fields makes sense on your end. So instead I would suggest adding a filter in Elementor Pro allowing end users to explicitly declare custom fields they want to access. Something like:

```
$valid_fields = apply_filters('elementor/custom_fields_allowed', []);
...
if ( '_' !== substr( $custom_key, 0, 1 ) || in_array( $custom_key, $valid_fields ) {
	$options[ $custom_key ] = $custom_key;
}
```

So the end user can do something like:

```
add_filter( 'elementor/custom_fields_allowed', function( $fields ) {
     $fields[] = '_job_description';
}); 
```

This has potential applications beyond WP Job Manager. Happy to issue a PR for that (though I think that chunk of code is in Pro, which may not be on GitHub for public contribution).

**Describe alternatives you've considered**
The alternative for problem 1 is the WP Job Manager folks set `'show_in_nav_menus' => true` when they register the job post type. Seems innocuous, but maybe not.

For the second issue, they would have to migrate to non-underscored keys, which would involve a pretty massive break to BC. Doubtful anything like that would ever occur.

Either way, some coordination between the two teams will probably be valuable.

**Comment by rkhoury82** [comment]
Elementor team - do you have any interest in helping with this one? 
The WP Jop Manager community is large and a lot of us are stuck using pre built themes - having this would be game changing !!

**Comment by Bilal-io** [comment]
It would be a great thing if Elementor team collaborated with WP Job Manager team, knowing that both plugins are well known. Please make this happen.

**Comment by TinyGiantStudios** [comment]
For what it's worth, we've created a dedicated connector plugin for WP Job Manager that'll add specific widgets to be used in Elementor. If you're keen, check out https://tinygiantstudios.co.uk/product/elementor-connector-for-wp-job-manager/

cc @rkhoury82 @Bilal-io

**Comment by gferguson1815** [comment]
Any progress on this or instructions on how we can get this working manually until it is included in a patch?

**Comment by Joe-Bloggs** [comment]
Is there any update on this?

It is quite hard to believe that this is still an ongoing issue.

**Comment by Joe-Bloggs** [comment]
> For what it's worth, we've created a dedicated connector plugin for WP Job Manager that'll add specific widgets to be used in Elementor. If you're keen, check out https://tinygiantstudios.co.uk/product/elementor-connector-for-wp-job-manager/
> 
@TinyGiantStudios does your plugin work with WPJM extensions too, specially Resumes?

**Comment by TinyGiantStudios** [comment]
@Joe-Bloggs - Apologies for the super, super late reply 😱

Our [Elementor Connector for WP Job Manager](https://tinygiantstudios.co.uk/product/elementor-connector-for-wp-job-manager/) only focuses on the main WPJM plugin although we are currently developing another connector for Resume Manager.

**Comment by zatakk** [comment]
Hi There. Is anyone working on this one. Can anyone give a work around atleast ?

**Comment by cestrian77** [comment]
Also looking for something like this too... seems bizarre there's not something out there that would allow the content to be easily accessible in Elementor. I'm trying to avoid adding another plugin too.

**Comment by welshlamb10** [comment]
Anyone have a fix for this plugin that displays companies to work with elementor?
https://github.com/Astoundify/wp-job-manager-companies
 
It loads a template called single-company.php, I want to override this so it loads the elementor theme builder templates I've used but there are no conditions I can set. 

I believe the companies are saved as custom meta (not taxonomies or custom fields)

I'm hoping someone here has a solution?

Thanks

**Comment by romanzy** [comment]
The plugin *[Fielder for Elementor](https://codecanyon.net/item/fielder-wordpress-custom-fields-for-elementor/36051940)* show custom fields anywhere on the page using Elementor editor.

**Comment by shehrozsheikh** [comment]
I recently developed this small and free plugin to handle this issue -> [Extension for WP Job Manager support in Elementor Pro](https://wordpress.org/plugins/extension-for-wp-job-manager-support-in-elementor-pro/)

Would love to hear your feedback on this.

**Comment by romanzy** [comment]
Hey Bro,

This is fantastic. Thank you for this.

Cheers,

Romanzy

**Comment by Nastin2022** [comment]
> I recently developed this small and free plugin to handle this issue -> [Extension for WP Job Manager support in Elementor Pro](https://wordpress.org/plugins/extension-for-wp-job-manager-support-in-elementor-pro/)
> 
> Would love to hear your feedback on this.

This is a lovely solution. However, why hasn't it been updated for the past 11 months? Is this extension still safe to use?

**Comment by shehrozsheikh** [comment]
@Nastin2022 Yes, it is. Just had to update the supported WordPress version so I just updated it in the readme file.
